src/main/java/nen/utils/Parser.java

```diff
@@ -0,0 +1,74 @@
+package nen.utils;
+
+import nen.commands.Command;
+import nen.commands.DoNothingCommand;
+import nen.exceptions.ArgumentMissingException;
+import nen.exceptions.InvalidInputException;
+
+/**
+ * This class represents a parser deals with making sense of the user command
+ * @author Gan Ren Yick (A0276246X)
+ */
+public class Parser {
+    private TaskList taskList;
+
+    /**
+     * Creates a parser which determined action of user input and make changes
+     * to taskList and give reaction to ui to print out
+     * @param taskList of tasks
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Parses given input and make changes to taskList and give reaction to ui to print out
+     * @param input by user
+     * @return a boolean which determined if nen2 should continue read user input
+     */
+    public Command parse(String input) {
+        String arg = "";
+        String action = input.split(" ")[0];
+        String react = "";
+        try {
+            switch(action) {
+            case "bye", "list", "find":
+                break;
+            case "mark", "delete", "unmark":
+                arg = String.valueOf(getIndex(input));
+                break;
+            default:
+                action = input;
+                break;
```

Comment:
Perhaps you can add cases for the task types here, e.g. "todo", "deadline", so that you can catch invalid commands here instead of having to catch it in the factory method of Tasks